src/main/java/com/codegym/jrugotom5/service/AdvertService.java

```diff
@@ -30,4 +35,21 @@ public List<AdvertDTO> getAdvertsByDateRange(LocalDate from, LocalDate to) {
                 .map(advert -> modelMapper.map(advert, AdvertDTO.class))
                 .collect(Collectors.toList());
     }
+
+    public AdvertDTO update(AdvertDTO advertDTO) {
+        Advert advert = advertRepository.findById(advertDTO.getId())
+                .orElseThrow(() -> new InvalidIdException("Advert not found with ID: " ));
+
+        Long userId = advertDTO.getUserId();
+        User user = userRepository.findById(userId)
+                .orElseThrow(() -> new InvalidIdException("User not found with ID: " ));
+
+        modelMapper.getConfiguration().setSkipNullEnabled(true);
+        modelMapper.map(advertDTO, advert);
```

Comment:
What the logic of your method?
As i understand, you:
1. get Advert from DB with DTO's "id";
2. get User from DB with DTO's "user_id";
3. change Advert (from 1.) with NON-NULL values from transfered DTO's fields;
4. update Advert in DB;
5. map Advert to DTO and return it;

I think rows 40-47 are not needed. You can just map DTO to Advert entity and update it in DB.
That is not your responcibility what kind of information client sends you. On the backend side you already have validation (with NotNull mandatory fields at DTO layer). 

Perhaps i would like to add check if `userId` from DTO relates to Advert as "Owner". But I think it's an authorization issue.

@Volodymyr95 ,what do you think?

Reply:
> I think rows 40-47 are not needed. You can just map DTO to Advert.class entity and update it in DB.

you cant, you may not get some values through endpoint, which means there is no need for them to be changed in db. but despite that you will map them as null to advert entity obj and when youll save this entity obj, those values will be overwriten with nulls in db. 2 options here: 1)like igor did; 2)make all values not-null in dto, so endpoint will require all of them, which is not very ideal.

Reply:
> ...but despite that you will map them as null to advert entity obj and when youll save this entity obj, those values will be overwriten with nulls in db...

So my point was that it isn't server responcibility. If client sends "null" - then it should be overwritten with null.
Try to remember RPG panel (from frontend project). When you change only 1 field, all unchanged fields will transfer to DB (by JS) 
Also there is no need to mark all fields as NotNull (just only those which are important for successfull processing).
But it's very interesting topic to discuss. Hope for Mentor's comments)